airflow-core/src/airflow/models/deadline.py

```diff
@@ -202,10 +202,20 @@ def handle_miss(self, session: Session):
         """Handle a missed deadline by running the callback in the appropriate host and updating the `callback_state`."""
         from airflow.sdk.definitions.deadline import AsyncCallback, SyncCallback
 
+        def get_simple_context():
+            from airflow.api_fastapi.core_api.datamodels.dag_run import DAGRunResponse
+
+            # TODO: Use the TaskAPI from within Triggerer to fetch full context instead of sending this context
+            #  from the scheduler
+            return {
+                "dag_run": DAGRunResponse.model_validate(self.dagrun).model_dump(mode="json"),
+                "deadline": {"id": self.id, "deadline_time": self.deadline_time},
+            }
+
         if isinstance(self.callback, AsyncCallback):
             callback_trigger = DeadlineCallbackTrigger(
                 callback_path=self.callback.path,
-                callback_kwargs=self.callback.kwargs,
+                callback_kwargs=(self.callback.kwargs or {}) | {"context": get_simple_context()},
```

Comment:
Nit: let's take anything called context from the user if they specify it? 
```suggestion
                callback_kwargs={"context": get_simple_context()} | (self.callback.kwargs or {}),
```

Reply:
I set context as a reserved kwarg so that shouldn't be a problem, otherwise they wouldn't get this provided context in the callback, potentially without even knowing it exists